source/MagicInternetBox.h

```diff
@@ -406,6 +408,16 @@ class MagicInternetBox {
 	 */
 	void failAllTask();
 
+	/**
+	 * Inform the host that a task requiring all members of the ship has succeeded
+	 */
+	void succeedAllTask();
+
+	/**
+	 * Inform the clients that the level has been won
+	 */
+	void forceWinLevel();
```

Comment:
Why is this necessary? Shouldn't each client have its own version of the state?

Reply:
Yes, but since we have tutorial levels that rely on events being completed (not time) and the host generates those events, the host needs to tell the clients when all is done.